### PR TITLE
[Merged by Bors] - 274 document the bundlebuilder and rego rule configmaps

### LINKED ISSUE #274: Document the BundleBuilder and Rego rule ConfigMaps
We currently do not explain the concept of the bundle builder (except two sentences in https://docs.stackable.tech/opa/stable/implementation-notes.html).
A Rego config map is shown in the getting started guide but not further explained. This should be better documented (in combination with the bundle builder).

## Acceptance

- [x] Implementation Notes / Concepts are extended to explain the bundle builder
  - [x] Extra Container with a local webserver, reading rego configmaps and providing a bundle.tar requested by the OPA container
  - [x] Pros (and Cons):
    - Pro: Bundles can be dynamically extendend / reduced by adding or removing rego config maps
    - Optional Cons: Depending on the amount of rules, zipping these can lead to high CPU consumption
- [x] Implementation Notes or Concepts are extended to explain the rego config maps
  - [x] Example rego rule configmaps are shown and explainend
  - [x] How this can be extended / dynamically adapted

### PR DESCRIPTION
**Pull request by razvan**
Fixes #274

### CODE REVIEW

File: docs/modules/ROOT/pages/implementation-notes.adoc [hunk 1]
```diff
@@ -12,5 +12,15 @@ We ensure local access via an https://kubernetes.io/docs/concepts/services-netwo
 
 == OPA Bundle Builder
 
-Each OPA pod runs a container that watches for `ConfigMap` objects labeled `opa.stackable.tech/bundle`.
-All `ConfigMap`s are bundled together in a single `bundle.tar.gz` file that is served to the colocated OPA container.
\ No newline at end of file
+Users can manage the Rego-rules evaluated by an OPA cluster by creating, updating and deleting `ConfigMap` resources.
+
+The role of the bundle builder is to convert these ressources to bundles (`tar.gz` files) and make it available as a HTTP endpoint.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/implementation-notes.adoc:17]:
```suggestion
The role of the bundle builder is to convert these resources to bundles (`tar.gz` files) and make them available via an HTTP endpoint.
```

File: docs/modules/ROOT/pages/implementation-notes.adoc [hunk 2]
```diff
@@ -12,5 +12,15 @@ We ensure local access via an https://kubernetes.io/docs/concepts/services-netwo
...
+The bundle builder runs in a side container of the OPA `Pod` as a simple HTTP server that OPA querying regularly
```
Comment (maltesander) [inline docs/modules/ROOT/pages/implementation-notes.adoc:18]:
```suggestion
The bundle builder runs in a side container of the OPA `Pod` as a simple HTTP server that OPA is querying regularly
```

File: docs/modules/ROOT/pages/implementation-notes.adoc [hunk 3]
```diff
@@ -12,5 +12,15 @@ We ensure local access via an https://kubernetes.io/docs/concepts/services-netwo
...
+(every 20 to 30 seconds) for updates.
+
+NOTE: Kubernetes limits the size of `ConfigMap`s to 1MB. Users have to take this limit into consideration when manging Rego-rules.
+
+Only `ConfigMap`s labeled with `opa.stackable.tech/bundle` are watched by the builder when updating bundles. The name of
+the `data` entries in the `ConfigMap` are used as file names when storing the rules in the bundle.
+
+NOTE: Currently it is the user's responsability to make sure these names do not collide.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/implementation-notes.adoc:26]:
```suggestion
NOTE: Currently it is the user's responsibility to make sure these names do not collide (as they will override each other).
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Can you please that that we use Rego/rego/Rego-rules/Rego rules consistently? I think sometimes i saw it written with lower case.

**Comment by razvan** [comment]
Replaced "rego" with "policy" where appropriate.

**Comment by razvan** [comment]
CI: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/opa-operator-it-custom/14/

File: docs/modules/ROOT/pages/implementation-notes.adoc (hunk 1 above)
Comment (maltesander) [reply docs/modules/ROOT/pages/implementation-notes.adoc]: The "ressources" is not fixed yet?

Comment (razvan) [reply docs/modules/ROOT/pages/implementation-notes.adoc]: I made the same changes locally that's why I couldn't apply your suggestion. It's fixed now.

File: docs/modules/getting_started/pages/first_steps.adoc [hunk 4]
```diff
@@ -16,9 +16,9 @@ include::example$code/getting_started.sh[tag=apply-opa-cluster]
 
 This will create an OPA cluster. The Operator deploys a DaemonSet, so every node in your cluster will have an OPA instance.
 
-== Deploy a rule
+== Deploy a policy rule
 
-Now deploy first link:https://www.openpolicyagent.org/docs/latest/policy-language/[Rego rule] to the Agent. Rules are deployed in ConfigMaps. Create a file `simple-rule.yaml` with the following contents:
+Now deploy first link:https://www.openpolicyagent.org/docs/latest/policy-language/[policy rule] to the Agent. Rules are deployed in ConfigMaps. Create a file `simple-rule.yaml` with the following contents:
```
Comment (maltesander) [inline docs/modules/getting_started/pages/first_steps.adoc:21]:
```suggestion
Now deploy the first link:https://www.openpolicyagent.org/docs/latest/policy-language/[policy rule] to OPA. Rules are deployed in ConfigMaps. Create a file `simple-rule.yaml` with the following contents:
```

**Review by maltesander — APPROVED** [review]
LGTM. One minor thing.

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**